apps/examples/testcase/le_tc/filesystem/tc_fs_procfs.c

```diff
@@ -94,10 +103,156 @@ static int read_dir_entries(const char *dirpath)
 
 	return ERROR;
 }
+#if !defined(CONFIG_DISABLE_MOUNTPOINT) && defined(CONFIG_FS_PROCFS)
+#ifndef CONFIG_FS_PROCFS_EXCLUDE_UPTIME
+static int procfs_uptime_ops(char *dirpath)
+{
+	int ret;
+	int	fd;
+	int fd1 = 0;
+	struct stat st;
+	fd = open(PROC_UPTIME_PATH, O_RDONLY);
+	if (fd < 0) {
+		printf("Failed to open \n" );
+		return ERROR;
+	}
+
+	ret = stat(PROC_UPTIME_PATH, &st);
+	if (ret != OK) {
+		printf("failed to stat \n");
+		return ERROR;
+	}
+	ret = dup2(fd, fd1);
+	if (ret != OK) {
+		printf("failed to duplicate the file data \n");
+		return ERROR;
+	}
+
+	ret = close(fd);
+	if (ret != OK) {
+		printf("failed to close \n");
+		return ERROR;
+		}
+
+	return OK;
+}
+#endif
+#endif
+
+#if !defined(CONFIG_DISABLE_MOUNTPOINT) && defined(CONFIG_FS_PROCFS)
+static int procfs_version_ops(char *dirpath)
+{
+	int ret;
+	int fd;
+	int fd1 = 0;
+	struct stat st;
+	fd = open(PROC_VERSION_PATH, O_RDONLY);
+	if (fd < 0) {
+		printf("Failed to open \n");
+		return ERROR;
+	}
+
+	ret = stat(PROC_VERSION_PATH, &st);
+	if (ret != OK) {
+		printf("failed to stat \n");
+		return ERROR;
+	}
+	ret = dup2(fd, fd1);
+	if (ret != OK) {
+		printf("failed to duplicate the file data \n");
+		return ERROR;
+	}
+
+	ret = close(fd);
+	if (ret != OK) {
+		printf("failed to close \n");
+		return ERROR;
+		}
+
+		return OK;
+}
+#endif
+static int procfs_rewind_tc(const char *dirpath)
+{
+	int count;
+	DIR *dir;
+	struct dirent *dirent;
+
+	dir = opendir(dirpath);
+	if (!dir) {
+		printf("Failed to open directory %s\n", dirpath);
+		return ERROR;
+	}
+
+	count = 0;
+	do {
+		dirent = readdir(dir);
+		count++;
+	} while (dirent != NULL);
+
+	rewinddir(dir);
+	do {
+		dirent = readdir(dir);
+		count--;
+	} while (dirent != NULL);
+
+	if (count != 0) {
+		printf("rewind operation failed %s \n",dirpath);
+		return ERROR;
+	}
+	closedir(dir);
+
+	return OK;
+}
+ void tc_fs_smartfs_mksmartfs(void)
+{
+	int ret;
+	ret = mksmartfs(SMARTFS_DEV_PATH, 1);
```

Comment:
Here you can check for 2nd definition also , in case of 
#ifdef CONFIG_SMARTFS_MULTI_ROOT_DIRS
int mksmartfs(FAR const char *pathname, uint8_t nrootdirs, bool force)